protocol/communities/manager.go

```diff
@@ -2473,7 +2519,10 @@ func (m *Manager) HandleCommunityRequestToJoin(signer *ecdsa.PublicKey, receiver
 		// If user is already a member, then accept request automatically
 		// It may happen when member removes itself from community and then tries to rejoin
 		// More specifically, CommunityRequestToLeave may be delivered later than CommunityRequestToJoin, or not delivered at all
-		acceptAutomatically := community.AcceptRequestToJoinAutomatically() || community.HasMember(signer)
+		// If community ownership changed, then accept request automatically
+		acceptAutomatically := community.AcceptRequestToJoinAutomatically() || community.HasMember(signer) ||
+			(existingRequestToJoin != nil && existingRequestToJoin.State == RequestToJoinStateOwnershipChanged)
```

Comment:
Could you please move it few lines up into the `if existingRequestToJoin != nil {` scope?